Source/DafnyLanguageServer/Workspace/IdeStateObserver.cs

```diff
@@ -80,6 +85,50 @@ public class IdeStateObserver : IObserver<IdeState> {
       notificationPublisher.PublishNotifications(LastPublishedState, snapshot).Wait();
 #pragma warning restore VSTHRD002
       LastPublishedState = snapshot;
+      logger.LogDebug($"Updated LastPublishedState to version {snapshot.Version}, uri {compilation.Uri}");
     }
   }
+
+  public void Migrate(DidChangeTextDocumentParams documentChange, int version) {
```

Comment:
It's interesting that you migrated the migration code in the IdeStateObserver. Could you compare the versions to ensure they are increasing? Is it necessary? I remember it was in other places like for gutter icons.

Reply:
The version number checks ensure that events from an older version don't publish information after events from a newer version.